canonical-web-and-design/jaas-dashboard
# Show all controllers in the controller lists

## Done

Previously controllers would only be shown in the controller list once they had connected. This branch renders all of the registered controllers whether they have connected or not.

This has not yet been designed so as an interim UI I've shown the websocket URL instead of the name and coloured it red to indicate an issue. The URL has a tooltip to indicate "disconnected". This will all be updated with a new UI in the near future.

Notes:
This register controller logic is needlessly complex and we will need to schedule some time to revisit this and clean it up.

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Register a controller with invalid data, see it go red.
- Bootstrap a controller on your cloud of choice and register it, see it work as it should.

## Details

Fixes #606 

## Screenshots

![Screen Shot 2020-09-02 at 4 45 58 PM](https://user-images.githubusercontent.com/532033/92045566-1a1bb480-ed3e-11ea-8f69-dfee1be30423.png)
